src/Currencies/ISOCurrencies.php

```diff
@@ -31,6 +32,25 @@ public function contains(Currency $currency)
         return isset(self::$currencies[$currency->getCode()]);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function getSubunitsFor(Currency $currency)
+    {
+        if (null === self::$currencies) {
+            self::$currencies = $this->loadCurrencies();
+        }
+
+        if (!isset(self::$currencies[$currency->getCode()])) {
+            throw new UnknownISOCurrencyException('Cannot find ISO currency '.$currency->getCode());
```

Comment:
UnkownCurrencyException would be better? Could be used by other repositories too.